federation/pkg/federation-controller/service/dns.go

```diff
@@ -44,6 +44,10 @@ func (s *ServiceController) getHealthyEndpoints(clusterName string, cachedServic
 			return nil, nil, nil, err
 		}
 		for _, ingress := range lbStatus.Ingress {
+			_, ok := cachedService.endpointMap[lbClusterName]
```

Comment:
We never set  cachedService.endpointMap[lbClusterName] == 0 in any logs, we only add it with value 1 or delete it from the map for the case no ready address.
I think this cloud be improved later, but now we'd keep it as is, sounds fair?